R/showLinePlots.R

```diff
@@ -0,0 +1,126 @@
+#' arranged and shown.
+#'
+#' @param vars A character vector. Usually just a single string. The variables
+#'   to be plotted. If \code{NULL} all rows from \code{data} are plotted.
+#' @param scales A single string. choose either \code{"free_y"} or \code{"fixed"}.
+#' @inheritParams showAreaAndBarPlots
+#' @return \code{NULL} is returned invisible.
+#' @examples
+#' \dontrun{
+#' showLinePlots(data, "Policy Cost|GDP Loss")
+#' }
+#' @export
+#' @importFrom rlang .data .env
+#' @importFrom dplyr bind_rows
+showLinePlots <- function(
+  data, vars = NULL, scales = "free_y",
+  mainReg = getOption("mip.mainReg")
+) {
+
+  # Validate function arguments.
+  stopifnot(is.quitte(data))
+  stopifnot(is.character(vars) || is.null(vars))
+  stopifnot(is.character(scales) && length(scales) == 1)
+  checkGlobalOptionsProvided("mainReg")
+  stopifnot(is.character(mainReg) && length(mainReg) == 1)
+
+  if (!is.null(vars)) {
+    data %>%
+      filter(.data$variable %in% .env$vars) %>%
+      droplevels() ->
+      d
+    label <- paste0(
+      paste0(vars, collapse = ","),
+      " [", paste0(levels(d$unit), collapse = ","), "]")
+  } else {
+    data %>%
+      droplevels() ->
+      d
+    label <- paste0(
+      paste0(levels(d$variable), collapse = ","),
+      " [", paste0(levels(d$unit), collapse = ","), "]")
+  }
+
+  d %>%
+    filter(.data$region == .env$mainReg, .data$scenario != "historical") %>%
+    droplevels() ->
+    dMainScen
+  d %>%
+    filter(.data$region == .env$mainReg, .data$scenario == "historical") %>%
+    droplevels() ->
+    dMainHist
+  d %>%
+    filter(.data$region != .env$mainReg, .data$scenario != "historical") %>%
+    droplevels() ->
+    dRegiScen
+  d %>%
+    filter(.data$region != .env$mainReg, .data$scenario == "historical") %>%
+    droplevels() ->
+    dRegiHist
+
+  if (!is.null(vars))
+    warnMissingVars(bind_rows(dMainScen, dRegiScen), vars)
+  if (NROW(dMainScen) == 0 && NROW(dRegiScen) == 0) {
+    warning("Nothing to plot.", call. = FALSE)
+    return(invisible(NULL))
+  }
+
+  if (NROW(dMainScen) == 0) {
+    p1 <- ggplot() + theme_minimal()
+  } else {
+    dMainScen %>%
+      mipLineHistorical(
+        x_hist = dMainHist,
+        ylab = label,
+        scales = scales,
+        plot.priority = c("x_hist", "x", "x_proj")
+      ) ->
+      p1
+  }
+  if (NROW(dRegiScen) == 0) {
+    p2 <- ggplot() + theme_minimal()
+  } else {
+    dRegiScen %>%
+      mipLineHistorical(
+        x_hist = dRegiHist,
+        ylab = NULL,
+        scales = scales,
+        plot.priority = c("x_hist", "x", "x_proj"),
+        facet.ncol = 3
+      ) ->
+      p2
+  }
+
+  # If a legend of the plots can be used as common legend for both plots,
+  # show that legend below mainReg-plot and only that legend.
+  mainHistModels <- levels(dMainHist$model)
+  regiHistModels <- levels(dRegiHist$model)
+  if (length(mainHistModels) == 0 || identical(mainHistModels, regiHistModels)) {
+    lgnd <- getLegend(p2)
```

Comment:
This abbreviation makes the code easier to write (unless you have autocompletion :wink: ) but harder to read. In my experience reading code is much harder to read than to write, so I believe most abbreviations are not a good idea.
```suggestion
    legend <- getLegend(p2)
```

Reply:
I think, this might be almost a kind of philosophical discussion. Let me present my arguments for `lgnd` and if you still prefer `legend` I will change it :)
1. `legend` conflicts with `graphics::legend()`.
2. Variables with a short, local scope should have short names. Shorter variables are easier to read, but their meaning usually is harder to understand or to remember. In this case, `lgnd <- getLegend(p2)` makes the meaning clear and it only has to be remembered for 10 lines of code.

(I totally agree that ease of writing code should not be an argument and that reading code is much harder than writing it.)

Reply:
The first point is a very good reason to not use `legend`, I was not aware of `graphics::legend`. The second point also makes perfect sense to me, although I disagree with `Shorter variables are easier to read`, unless names are super long. I find `legend` much easier to read than `lgnd`. I agree `lgnd <- getLegend(p2)` makes it very clear though, so I'm fine with `lgnd` here :+1: